Mathlib/Algebra/BigOperators/Multiset/Basic.lean

```diff
@@ -154,6 +154,16 @@ theorem pow_count [DecidableEq α] (a : α) : a ^ s.count a = (s.filter (Eq a)).
 #align multiset.pow_count Multiset.pow_count
 #align multiset.nsmul_count Multiset.nsmul_count
 
+@[to_additive]
+theorem filter_prod_mul_sub_filter_prod_eq_prod (p) [DecidableEq α] [DecidablePred p] :
+    (s.filter p).prod * (s - (s.filter p)).prod = s.prod := by
+  rw [sub_filter_eq_filter_not]
```

Comment:
I think this would be better without the `-`:
```suggestion
    (s.filter p).prod * (s.filter (fun a ↦ ¬ p a)).prod = s.prod := by
```

I would guess you can prove it more concisely with a lemma that says `(s + t).prod = s.prod * t.prod`

Reply:
Nice! Indeed!